### ISSUE TITLE
Action only via relationship not work with dependent field

### ISSUE THREAD

**Issue by et-hoangdv** [laravel/nova-issues#6575]
I have `Event` and `Attendee` resources which have many to many relationship.

I have an action of `Attendee` that need to only show up on `Event` resource detail page.

```php
class Attendee extends Resource
{
public function actions(NovaRequest $request): array
    {
        return [
            (new SendMailByTemplate)->showInline()->canSee(function (NovaRequest $request) {
                return $request->viaRelationship();
            }),
        ];
    }
}
```

```php
class SendMailByTemplate extends Action
{
    public function fields(NovaRequest $request)
    {
        return [
            Select::make(__('Template'), 'template')
                ->options(...),

            Text::make(__('Subject'), 'subject')
                ->dependsOn(
                    ['template'],
                    function (Text $field, NovaRequest $request, FormData $formData) use ($mailTemplates) {
                        if ($formData->template){
                            // set value
                        }
                    }
                ),
        ];
    }
}
```

This is work fine until I change the select option of the `template` field which it will call an "action API" which not pass params like `viaResource`, `viaResourceId`, `viaRelationship` which make the `canSee` return `false` everytime.

I have temporarily fixed it by change the `canSee` condition but I don't know if that a right way.

```php
(new SendMailByTemplate)->showInline()->canSee(function (NovaRequest $request) {
                return $request->viaRelationship() || !empty($request->resources);
            }),
```

- Laravel Version: 11
- Nova Version: 4

**Comment by crynobone** [comment]
Can you explain the requirement you used for `canSee()` and some screenshot to explain the issue better?

**Comment by et-hoangdv** [comment]
The requirement here is to send email according to the template to the attendees for each event. So the action of sending email according to the template should only appear on the event detail screen, not the attendees screen.
so I set this action to only appear when there is a relationship.

1. I go to the event detail page which list all attendees join that event.
![Screenshot 2024-11-06 094455](https://github.com/user-attachments/assets/94054a91-aaae-4793-8e89-f23b773f69a4)

2. I select some users and then select the action to send email according to the template.
![Screenshot 2024-11-06 094605](https://github.com/user-attachments/assets/409c9c4d-8de7-413f-abb3-c0a01e6468a4)

3. I choose email template. Based on the template type I will get data in DB and fill in input `Subject` and `Content`. This is also when error happen.
![Screenshot 2024-11-06 094618](https://github.com/user-attachments/assets/868497e8-5b97-4a2e-b250-5f7dcb462ac4)

When I select template it will call the action API based on the number of dependent fields I have. And importantly it doesn't pass the other parameters `viaResource`, `viaResourceId`, `viaRelationship`  so it causes a 404 error at `canSee`
![Screenshot 2024-11-06 094808](https://github.com/user-attachments/assets/2d4c01cc-f331-4b9f-9a57-0252ca3cc74f)

**Comment by crynobone** [comment]
Yes it doesn't provide those information because it was not designed to be use in such way.
